src/content/1.7/modules/symfonyform/_index.md

```diff
@@ -0,0 +1,264 @@
+
+```php
+class DemoConfigurationChoiceFormDataProvider implements FormDataProviderInterface
+{
+    /** If you want to prefill your from with data when it's loaded(previous values if it's setting form) you can return that data here */
+    public function getData(): array
+    {
+        return $this->getData();
+    }
+
+    public function setData(array $data): array
+    {
+        $this->saveData($data);
+        /** You can return array of errors if something gfoes wrong */
+        return ['Something went wrong'];
+
+    }
+}
+```
+
+##### CRUD Form
+
+To handle CRUD form PrestaShop uses ``PrestaShop\PrestaShop\Core\Form\IdentifiableObject\Handler``
+You can use ``FormHandlerFactory`` ``create`` function in order to create the handler. Primary argument you need here is DataHandler which would be responsible for saving your data.
+
+It needs to implement ``FormDataHandlerInterface`` and can be passed to your form handler in a service.
+
+```yaml
+  prestashop.core.form.identifiable_object.handler.feature_form_handler:
+    class: 'PrestaShop\PrestaShop\Core\Form\IdentifiableObject\Handler\FormHandler'
+    factory: 'prestashop.core.form.identifiable_object.handler.form_handler_factory:create'
+    arguments:
+      #  Your data handler
+      - '@prestashop.core.form.identifiable_object.data_handler.feature_form_data_handler'
+```
+
+Data handler has to functions ``create`` and ``update`` where you need to define logic for your object. 
+
+In controller use either ``FormHandler->handle`` or ``FormHandler->handleFor``.
+
+``handleFor`` requires you to have pass an ID as it's used to update object, while ``handle`` is used to create one.
+
+
+##### Using different form action
+In some cases you want to use new action for form saving (for example if you have multiple forms in one page and you want each to have separate action)
+In that case you can define route in template file, you need to modify a ``form_start`` and add path to your action.
+```twig
+ {{ form_start(demoConfigurationChoiceForm, {attr : {class: 'form'}, action: path('demo_configuration_choices_form_save') }) }}
+```
+
+In that case when you are done handling the form you will need to redirect back to original form. You can pass success or error messages using controller function
+``addFlash``. You can't use constraints when handling form like this since after redirecting you won't see any error messages. ``$textForm->isValid()`` will still return negative however so it's advise to not to use it, validate during form handling and add errors using ``addFlash``.  
+##### 
+
+```php
+<?php
+header('Expires: Mon, 26 Jul 1997 05:00:00 GMT');
+header('Last-Modified: ' . gmdate('D, d M Y H:i:s') . ' GMT');
+header('Cache-Control: no-store, no-cache, must-revalidate');
+header('Cache-Control: post-check=0, pre-check=0', false);
+header('Pragma: no-cache');
+header('Location: ../');
+exit;
+```
```

Comment:
This is the default behavior of Symfony redirection. No need to handle it manually.